### ISSUE TITLE
Create a Trainer class for learning rate scheduling

### ISSUE THREAD

**Issue by yoptar** [deepmipt/DeepPavlov#800]
For now learning rate scheduling is implemented using the [LRScheduledModel](https://github.com/deepmipt/DeepPavlov/blob/613d265f7371ba05365a7d44485066293c169674/deeppavlov/core/models/lr_scheduled_model.py#L129-L132) class and its inheritors. It would be great to move this logic to a trainer class inherited from the [NNTrainer](https://github.com/deepmipt/DeepPavlov/blob/613d265f7371ba05365a7d44485066293c169674/deeppavlov/core/trainers/nn_trainer.py#L33-L38) class.

**Comment by vikmary** [comment]
LRScheduledModel performs
- learning rate and momentum updating
- fitting of learning rate on a small subset of data (before main phase of training), see [LRScheduledModel.fit](https://github.com/deepmipt/DeepPavlov/blob/613d265f7371ba05365a7d44485066293c169674/deeppavlov/core/models/lr_scheduled_model.py#L294) for details.

**Comment by nikolay-bushkov** [comment]
FYI
https://www.tensorflow.org/alpha/guide/keras/custom_callback#learning_rate_scheduling
